modules/pagerduty/pagerduty.go

```
@@ -0,0 +1,188 @@
+					log.Printf("[dryrun] pagerduty %s: would have deleted user %s", r.p.Subdomain, user.Email)
+				} else {
+					err := client.DeleteUser(user.ID)
+					if err != nil {
+						log.Fatalf("[error] pagerduty %s: could not delete user %s: %s", r.p.Subdomain, user.Email, err.Error())
+					}
+				}
+				r.notify(ldapEmail, fmt.Sprintf("Userplex deleted %s from Pagerduty account %s", ldapEmail, r.p.Subdomain))
+			}
+		}
+	}
+
+	for ldapEmail, isInPagerduty := range ldapers {
+		// if there's an ldapuid/localuid mapping for the user
+		var (
+			pagerdutyEmails       []string
+			pagerdutyEmailsString string
+		)
+		if emails, ok := r.ldapEmailToPagerdutyEmails[ldapEmail]; ok {
+			pagerdutyEmails = emails
+			pagerdutyEmailsString = strings.Join(pagerdutyEmails, ", ") + " (pagerduty) / "
+		}
+
+		if !isInPagerduty && r.Conf.Create {
+			log.Printf("[info] pagerduty %s: user %s%s (ldap) is not in Pagerduty account %s", r.p.Subdomain, pagerdutyEmailsString, ldapEmail, r.p.Subdomain)
+			if !r.Conf.ApplyChanges {
+				log.Printf("[dryrun] pagerduty %s: would have created user %s", r.p.Subdomain, ldapEmail)
+			} else {
+				// add user to pagerduty
+				client.CreateUser(pagerduty.User{
+					Email: ldapEmail,
+					Name:  ldapEmail,
+				})
+			}
+			r.notify(ldapEmail, fmt.Sprintf("Userplex added %s to Pagerduty account %s", ldapEmail, r.p.Subdomain))
+		}
+	}
+
+	return
+}
+
+func (r *run) buildLdapMapping() {
+	r.pagerdutyEmailToLdapEmail = make(map[string]string)
+	r.ldapEmailToPagerdutyEmails = make(map[string][]string)
+	for _, mapping := range r.Conf.UidMap {
+		r.pagerdutyEmailToLdapEmail[mapping.LocalUID] = mapping.LdapUid
+		r.ldapEmailToPagerdutyEmails[mapping.LdapUid] = append(r.ldapEmailToPagerdutyEmails[mapping.LdapUid], mapping.LocalUID)
+	}
+}
+
+func (r *run) getLdaperEmails() (ldapUsers map[string]bool) {
+	ldapUsers = make(map[string]bool)
+	users, err := r.Conf.LdapCli.GetEnabledUsersInGroups(r.Conf.LdapGroups)
+	if err != nil {
+		return
```

You're silencing this error by not returning it. Intentional?

---

Fixed